project4/name_server/bootstrap_driver.h

```diff
@@ -0,0 +1,28 @@
+/**
+ * @file Bootstrap Driver
+ */
+#ifndef PROJECT4_BOOTSTRAP_DRIVER_H
+#define PROJECT4_BOOTSTRAP_DRIVER_H
+
+#include "../../common/thread_pool/task.h"
+#include "../../common/thread_pool/thread_pool.h"
+
+#include "tasks/console_task.h"
+#include "tasks/bootstrap_task.h"
+
+namespace nameserver {
+
+class BootstrapDriver {
```

Comment:
Is it possible to unify this class with `NameserverDriver`? You could use DI to inject the actual `Nameserver` subclass that you want to manage. If not, I would recommend creating a common interface class that both `BootstrapDriver` and `NameserverDriver` inherit from.

Reply:
I have to say, I'm not familiar with DI (dependency injection?). It may be easier to make a generic driver interface and have the `BootstrapDriver` and the `NamespaceDriver` inherit it.

Reply:
Dependency injection, yes. It sounds fancy, but in this case, the idea is actually pretty simple. Mostly, it would amount to adding an argument to the constructor for `BootstrapDriver` that allowed you to pass a `Nameserver` subclass, which would then be used by the driver. I'm also a fan of this design pattern also because it makes testing easier. My personal opinion is that in this case, it will also reduce code duplication more than writing an interface.